runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -1232,38 +1223,22 @@ def DeleteCommands()
   delcommand Var
   delcommand Winbar
 
-  if k_map_saved isnot null_dict
-    if !empty(k_map_saved) && k_map_saved.buffer
-      # pass
-    elseif !empty(k_map_saved) && !k_map_saved.buffer
-      nunmap K
-      mapset(k_map_saved)
-    elseif empty(k_map_saved)
-      silent! nunmap K
-    endif
-    k_map_saved = null_dict
-  endif
-  if plus_map_saved isnot null_dict
-    if !empty(plus_map_saved) && plus_map_saved.buffer
-      # pass
-    elseif !empty(plus_map_saved) && !plus_map_saved.buffer
-      nunmap +
-      mapset(plus_map_saved)
-    elseif empty(plus_map_saved)
-      silent! nunmap +
-    endif
-    plus_map_saved = null_dict
-  endif
-  if minus_map_saved isnot null_dict
-    if !empty(minus_map_saved) && minus_map_saved.buffer
-      # pass
-    elseif !empty(minus_map_saved) && !minus_map_saved.buffer
-      nunmap -
-      mapset(minus_map_saved)
-    elseif empty(minus_map_saved)
-      silent! nunmap -
-    endif
-    minus_map_saved = null_dict
+  if !empty(saved_k_map)
```

Comment:
Yes. The logic now is much easier and it goes like the following:

1. At termdebug startup user-defined mapping are always saved no matter of the user settings. Maps are always saved even if they are empty (that happens in the Init function).
2. Termdebug set the mapping depending on user settings.
3. At teadown, termdebug restore the saved mapping exactly as they were before staring gdb.

The empty condition is a guard in case the `saved_k_map` is empty. In that case `mapset` would return an error.

Reply:
But the problem is, if the mapping was a buffer-local (the `buffer` key is set) mapping, we may restore it wrongly here (because it get's re-created in the wrong buffer).

Reply:
Done! Check it now. I also updated the related variables names with capital letter for `K` since it is `K` mapping and not `k`.